Switching from FileSize to DataSize for logback size-based properties.

nosan opened the pull request:
see gh-15930.
`DataSize` can contain a `negative` value, Should we handle this? If yes, should we throw an exception or use a default value?
Thanks

snicoll commented:
@nosan thanks for the PR. Did you check if the `FileSize` string format works transparently with `DataSize`?

nosan commented:
Thanks @snicoll,
I have added some tests to check this [here](https://github.com/spring-projects/spring-boot/pull/15957/files#diff-cc08d614c5d3cabca80135c77e110313R401) and [here](https://github.com/spring-projects/spring-boot/pull/15957/files#diff-cc08d614c5d3cabca80135c77e110313R491).

Current implementation parses the value via `DataSize` class, if something wrong, then it tries to parse a value using `FileSize` class.

**P.S.** one case that which is not covered is a `negative` value. I think the exception should be thrown as a value is not valid.

nosan commented:
FYI @snicoll,
I changed the implementation this morning and added more tests.  Instead of `stupid` parsing via both classes, I added a regex to handle both formats.

snicoll commented:
@nosan than you for the PR but I've decided to go with a different approach, see https://github.com/spring-projects/spring-boot/commit/eee07efee1bb6a9bdfd6e34000eb91b7cf347502 for more details.

nosan commented:
@snicoll sure, no problem.
BTW,  just to make sure, you did nothing with a negative value, is it ok?

snicoll commented:
I noticed your comment about that. Yes I think it is ok as the main purpose was to align tooling and size format
